src/universal/components/ReflectionEditorWrapper.js

```diff
@@ -0,0 +1,211 @@
+    const {handleChange, setEditorState} = this.props;
+    if (this.entityPasteStart) {
+      const {anchorOffset, anchorKey} = this.entityPasteStart;
+      const selectionState = editorState.getSelection().merge({
+        anchorOffset,
+        anchorKey
+      });
+      const contentState = entitizeText(editorState.getCurrentContent(), selectionState);
+      this.entityPasteStart = undefined;
+      if (contentState) {
+        setEditorState(EditorState.push(editorState, contentState, 'apply-entity'));
+        return;
+      }
+    }
+    if (editorState.getSelection().getHasFocus() && handleChange) {
+      handleChange(editorState);
+    }
+    setEditorState(editorState);
+  };
+
+  handleReturn = (e) => {
+    const {handleReturn} = this.props;
+    if (handleReturn) {
+      return handleReturn(e);
+    }
+    return 'not-handled';
+  };
+
+  handleKeyCommand = (command) => {
+    const {handleKeyCommand} = this.props;
+    if (handleKeyCommand) {
+      return handleKeyCommand(command);
+    }
+    return undefined;
+  };
+
+  keyBindingFn = (e) => {
+    const {keyBindingFn} = this.props;
+    if (keyBindingFn) {
+      keyBindingFn(e);
+    }
+    if (e.key === 'Escape') {
+      e.preventDefault();
+      this.removeModal();
+    } else {
+      getDefaultKeyBinding(e);
+    }
+  };
+
+  handleBeforeInput = (char) => {
+    const {handleBeforeInput} = this.props;
+    if (handleBeforeInput) {
+      return handleBeforeInput(char);
+    }
+    return undefined;
+  }
+
+  handlePastedText = (text) => {
+    if (text) {
+      for (let i = 0; i < textTags.length; i++) {
+        const tag = textTags[i];
+        if (text.indexOf(tag) !== -1) {
+          const selection = this.props.editorState.getSelection();
+          this.entityPasteStart = {
+            anchorOffset: selection.getAnchorOffset(),
+            anchorKey: selection.getAnchorKey()
+          };
+        }
+      }
+    }
+    return 'not-handled';
+  };
+
+  render() {
+    const {ariaLabel, editorState, hasDragLock, isBlurred, onBlur, onFocus, placeholder, readOnly} = this.props;
+    const userSelect = readOnly ? 'none' : 'text';
+    return (
+      <EditorStyles hasDragLock={hasDragLock} isBlurred={isBlurred}>
+        <Editor
+          ariaLabel={ariaLabel}
+          blockStyleFn={this.blockStyleFn}
+          editorState={editorState}
+          handleBeforeInput={this.handleBeforeInput}
+          handleKeyCommand={this.handleKeyCommand}
+          handlePastedText={this.handlePastedText}
+          handleReturn={this.handleReturn}
+          keyBindingFn={this.keyBindingFn}
+          onBlur={onBlur}
+          onFocus={onFocus}
+          onChange={this.handleChange}
+          placeholder={placeholder}
+          readOnly={readOnly}
+          ref={this.setEditorRef}
+          style={{padding: '0.8rem', userSelect, WebkitUserSelect: userSelect}}
+        />
+      </EditorStyles>
+    );
+  }
+}
+
+export default withSuggestions(
```

Comment:
reflections can have links & font changes, but no mentions or hashtags.

Reply:
Bold, italic and the like didn't work. Neither did the `:` for emoji. Is that intentional? If so, it's fine for now.